add traceability adr (#95)
factory-x-contributions/architecture-decisions: docs/hercules_use_case_adr/adr106-traceability/README.md

### Data Consumer

The Data Consumer MUST expose the endpoints according to the following Architecture Decision Records (ADRs):

| Architecture Decision Record (ADR)                  | Version | Link                                                   | 
| --------------------------------------------------- | ------- | ------------------------------------------------------ | 
| ADR 002 – Cross-Company Authorization and Discovery | 0.2.0   | https://factory-x-contributions.github.io/architecture-decisions/docs/hercules_network_adr/adr002-authorization-discovery | 
| ADR 003 – Authentication for Dataspaces             | 0.2.0   | https://factory-x-contributions.github.io/architecture-decisions/docs/hercules_network_adr/adr003-authentication | 

## Data Models
### Submodels

Dependent on the scenario that should be supported, different data models MUST or MAY be used to support the Traceabily use case.  
Multiple scenarios can be supported, but at least scenario 1 MUST be supported.

#### Scenario 1 - Catena-X Quality Notifications

> The Traceability Use Case is based on the results of the project Catena-X. This includes the support for quality notifications between partners in the supply chain.

Reviewer: ```suggestion
The Traceability Use Case is based in the Catena-X dataspace standards. This includes the support for quality notifications between partners in the supply chain.
```
